### PR TITLE
Add e2e test for VPA validation

### PR DESCRIPTION
**Pull request by bskiba**

### CODE REVIEW

File: vertical-pod-autoscaler/e2e/v1/admission_controller.go [hunk 1]
```diff
@@ -449,6 +449,47 @@ var _ = AdmissionControllerE2eDescribe("Admission-controller", func() {
...
+					"kind": "Deployment",
+					"name":"hamster"
+				},
+		   	"resourcePolicy": {
+		  		"containerPolicies": [{"containerName": "*", "minAllowed":{"cpu":"50m"}}]
+		  	}
+		  }
+		}`)
+		err := InstallRawVPA(f, validVPA)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred(), "Valid VPA object rejected") //
```
Comment (jbartosik) [inline vertical-pod-autoscaler/e2e/v1/admission_controller.go]: Empty comment

File: vertical-pod-autoscaler/e2e/v1/common.go [hunk 2]
```diff
@@ -250,6 +250,17 @@ func InstallVPA(f *framework.Framework, vpa *vpa_types.VerticalPodAutoscaler) {
 	gomega.Expect(err).NotTo(gomega.HaveOccurred(), "unexpected error creating VPA")
 }
 
+// InstallRawVPA installs a VPA object passed in as raw json in the test cluster.
+func InstallRawVPA(f *framework.Framework, obj interface{}) error {
+	vpaClientSet := getVpaClientSet(f)
+	err := vpaClientSet.AutoscalingV1().RESTClient().Post().
```
Comment (jbartosik) [inline vertical-pod-autoscaler/e2e/v1/common.go]: Nit: return vpaClientSet.AutoscalingV1().... ?

File: vertical-pod-autoscaler/e2e/v1/admission_controller.go [hunk 3]
```diff
@@ -449,6 +449,47 @@ var _ = AdmissionControllerE2eDescribe("Admission-controller", func() {
...
+
+		ginkgo.By("Setting up invalid VPA object")
+		invalidVPA := []byte(`{
```
Comment (jbartosik) [inline vertical-pod-autoscaler/e2e/v1/admission_controller.go]: Nit: difference would be easier to find if both were formatted the same way

File: vertical-pod-autoscaler/e2e/v1/admission_controller.go [hunk 4]
```diff
@@ -449,6 +449,47 @@ var _ = AdmissionControllerE2eDescribe("Admission-controller", func() {
...
+			"apiVersion": "autoscaling.k8s.io/v1",
+			"metadata": {"name": "hamster-vpa-invalid"},
+			"spec": {
+				"targetRef": {
+		    	"apiVersion": "apps/v1",
+		    	"kind": "Deployment",
+		    	"name": "hamster"
+		   	},
+		   	"resourcePolicy": {
+		  		"containerPolicies": [{"containerName": "*", "minAllowed":{"requests":{"cpu":"50m"}}}]
```
Comment (jbartosik) [inline vertical-pod-autoscaler/e2e/v1/admission_controller.go]: I think `minAllowed` is the only difference between the valind and the invalid VPAs. Please leave a comment pointing to the difference

File: vertical-pod-autoscaler/e2e/v1beta1/admission_controller.go [hunk 5]
```diff
@@ -474,6 +474,47 @@ var _ = AdmissionControllerE2eDescribe("Admission-controller", func() {
 		}
 	})
 
+	ginkgo.It("accepts valid and rejects invalid VPA object", func() {
```
Comment (jbartosik) [inline vertical-pod-autoscaler/e2e/v1beta1/admission_controller.go]: Same comments as with V1

File: vertical-pod-autoscaler/e2e/v1beta2/admission_controller.go [hunk 6]
```diff
@@ -449,6 +449,47 @@ var _ = AdmissionControllerE2eDescribe("Admission-controller", func() {
 		}
 	})
 
+	ginkgo.It("accepts valid and rejects invalid VPA object", func() {
```
Comment (jbartosik) [inline vertical-pod-autoscaler/e2e/v1beta2/admission_controller.go]: Same comments as with V1

File: vertical-pod-autoscaler/e2e/v1/admission_controller.go (hunk 3 above)
Comment (bskiba) [reply vertical-pod-autoscaler/e2e/v1/admission_controller.go]: Done

File: vertical-pod-autoscaler/e2e/v1/admission_controller.go (hunk 4 above)
Comment (bskiba) [reply vertical-pod-autoscaler/e2e/v1/admission_controller.go]: Done

File: vertical-pod-autoscaler/e2e/v1/common.go (hunk 2 above)
Comment (bskiba) [reply vertical-pod-autoscaler/e2e/v1/common.go]: I'd rather not, this way it is clear that i Do() the request and then return err.Error()

Comment (jbartosik) [reply vertical-pod-autoscaler/e2e/v1/common.go]: Ack

**Pull request merged**